src/System.Text.Json/tests/Serialization/DictionaryTests.cs

```diff
@@ -829,6 +829,34 @@ public static void DeserializeUserDefinedDictionaryThrows()
             Assert.Throws<NotSupportedException>(() => JsonSerializer.Parse<IImmutableDictionaryWrapper>(json));
         }
 
+        [Fact]
```

Comment:
If https://github.com/dotnet/corefx/issues/38557 is also fixed here, consider adding tests cases from that issue.

Reply:
Added regression tests.